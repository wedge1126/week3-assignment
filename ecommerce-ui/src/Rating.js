import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';

export default class HomeCard extends React.Component {
  render() {
      const stars = this.props.value.stars;
      const reviews = this.props.value.reviews;
      let starIcons = [];
      let key = 0;
      for(let i = 0; i < Math.floor(stars); ++i) {
        starIcons.push(<FontAwesomeIcon className="checked" icon={['fas', 'star']} key={++key} />)
      }
      if(Math.floor(stars) !== Math.ceil(stars)) {
        starIcons.push(<FontAwesomeIcon className="checked" icon={['fas', 'star-half-alt']} key={++key} />)
      }
      for(let i = 0; i < 5 - Math.ceil(stars); ++i) {
        starIcons.push(<FontAwesomeIcon className="checked" icon={['far', 'star']} key={++key} />)
      }

      return <div className="rating">
        <span className="rating-stars">{starIcons}</span>
        <span className="rating-reviews">{reviews}</span>
      </div>
  }
}

HomeCard.propTypes = {
    value: PropTypes.shape({
        stars: PropTypes.number.isRequired,
        reviews: PropTypes.number.isRequired
    }).isRequired
}