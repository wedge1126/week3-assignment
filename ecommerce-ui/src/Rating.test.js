import React from 'react';
import { shallow } from 'enzyme';

import Rating from './Rating';

describe('Rating', () => {
    it('should render 5 stars correctly', () => {
        const rating = { stars: 5, reviews: 1 };
        const component = shallow(<Rating value={rating} />);
        
        expect(component).toMatchSnapshot();
    });

    it('should render 2.3 stars correctly', () => {
        const rating = { stars: 2.3, reviews: 1 };
        const component = shallow(<Rating value={rating} />);
        
        expect(component).toMatchSnapshot();
    });

    it('should render 1 star correctly', () => {
        const rating = { stars: 1, reviews: 1 };
        const component = shallow(<Rating value={rating} />);
        
        expect(component).toMatchSnapshot();
    });

    it('should render 0 stars correctly', () => {
        const rating = { stars: 0, reviews: 1 };
        const component = shallow(<Rating value={rating} />);
        
        expect(component).toMatchSnapshot();
    });
});