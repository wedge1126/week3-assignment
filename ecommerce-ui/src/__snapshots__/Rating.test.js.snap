// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render 0 stars correctly 1`] = `
<div
  className="rating"
>
  <span
    className="rating-stars"
  >
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="1"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="2"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="3"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="4"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="5"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <span
    className="rating-reviews"
  >
    1
  </span>
</div>
`;

exports[`Rating should render 1 star correctly 1`] = `
<div
  className="rating"
>
  <span
    className="rating-stars"
  >
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="1"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="2"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="3"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="4"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="5"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <span
    className="rating-reviews"
  >
    1
  </span>
</div>
`;

exports[`Rating should render 2.3 stars correctly 1`] = `
<div
  className="rating"
>
  <span
    className="rating-stars"
  >
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="1"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="2"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star-half-alt",
        ]
      }
      inverse={false}
      key="3"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="4"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "far",
          "star",
        ]
      }
      inverse={false}
      key="5"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <span
    className="rating-reviews"
  >
    1
  </span>
</div>
`;

exports[`Rating should render 5 stars correctly 1`] = `
<div
  className="rating"
>
  <span
    className="rating-stars"
  >
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="1"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="2"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="3"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="4"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="checked"
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "star",
        ]
      }
      inverse={false}
      key="5"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <span
    className="rating-reviews"
  >
    1
  </span>
</div>
`;
